graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/SearchDTO.java

```diff
@@ -41,7 +42,7 @@ public abstract class SearchDTO {
     public abstract String id();
 
     @JsonProperty
-    public abstract Set<QueryDTO> queries();
+    public abstract ImmutableSet<QueryDTO> queries();
```

Comment:
ImmutableSet does not guarantee maintaining insertion order. In fact it has some sub classes that will not maintain it at all (i.e. ImmutableSortedSet will sort the entries). 
While in this particular case it works, I'd consider replacing it with LinkedHashSet, to explicitly state that we are interested in maintaining the insertion order of the set.